components/job-orchestration/job_orchestration/scheduler/scheduler_data.py

```diff
@@ -34,9 +35,16 @@ class CompressionTaskSuccessResult(CompressionTaskResult):
     total_compressed_size: int
 
 
-class SearchJob:
-    def __init__(self, async_task_result: any) -> None:
-        self.async_task_result: any = async_task_result
+class SearchSubJob(BaseModel):
```

Comment:
With the way you've structured the code, I don't think we actually need this data structure. Instead, we could change `SearchJob.current_sub_job` to `SearchJob.sub_job_async_task_result`?

Reply:
Right